### ISSUE TITLE
Raspi imager changes

### ISSUE THREAD

**Issue by mstormi** [openhab/openhabian#1837]
See below, just received that today. 

@Larsen-Locke would you be able to adapt that on short notice ? Note the deadline. of Feb 7.

```
As part of the most recent release of Raspberry Pi Imager (v1.8.1), we've introduced two major changes:

1) A requirement for OS providers to supply a point of contact to Raspberry Pi, so that we can get in touch about future schema changes, hardware releases, etc.
2) A mechanism for flagging image compatibility with Raspberry Pi devices.

Point of contact

...

Device filtering

This mechanism requires annotating your provided OS list json file with a `devices` tag, that contains values matching those in our root imager JSON file's list (available at: https://downloads.raspberrypi.org/os_list_imagingutility_v3.json)

For example, to annoate for Raspberry Pi 4 and 5, you would need to add the following array to your os_list items:

"devices": ["pi5-64bit", "pi4-64bit"]

Note that this filtering mechanism is being introduced on a rolling basis. Today, Raspberry Pi 5's device filter will not show unannotated OSes. I expect to expand this exclusive filtering to previous Raspberry Pi devices, and expect to contact the named point of contact to coordinate their operating system's annoations.
```

**Comment by Larsen-Locke** [comment]
yes, of course, I already looked at it. I'll edit the template today.
What about my PR with the username-bugfix. Can you merge it?

**Comment by mstormi** [comment]
> What about my PR with the username-bugfix. Can you merge it?

seems you mixed raspi-imager changes into that username bugfix ?

**Comment by Larsen-Locke** [comment]
ups, I made a new branch after the changes and then cherry-picked the last commit. Didn't know that the commit remains in the main branch (still learning).
You're going to merge anyway? All changes should be save.

**Comment by Larsen-Locke** [comment]
But we need a release before Feb 7th, so that we have the json with the required tag online under the path https://github.com/openhab/openhabian/releases/latest/download/rpi-imager-openhab.json.

**Comment by Larsen-Locke** [comment]
The PR for only the Raspberry-imager is 
https://github.com/openhab/openhabian/pull/1839

**Comment by udo1toni** [comment]
FYI, RPi Imager does not work with latest releasse (404, which is correct, as the files linked in rpi-imager-openhab.json are different to the current images.)

**Comment by Larsen-Locke** [comment]
I check it out

**Comment by Larsen-Locke** [comment]
The filenames of the images have changed, I guess manually (e.g. openhabian-64-latest-202402041906.img.xz). 
build.bash seems unchanged
@mstormi : Do you have an idea why and where that was changed?

**Comment by mstormi** [comment]
@Larsen-Locke do we still have a TODO here?

**Comment by Larsen-Locke** [comment]
yes we do. I asked you how to proceed for bookworm but you didn't answer. When I know I can change the build-script and the template accordingly.
This was the question:
"How do you want to change the naming?
The filenames are written in build.bash line 399 followed by the generation of the json. I wouldn’t do renaming somewhere else (like in the yaml). What about adding “oldstable” or “latest” to the current filename. Like
openhabian-pi-raspios32-latest-202308121256-gitc6a2fa1-crcc361b879.img.xz?"

**Comment by mstormi** [comment]
it was an awfully time consuming lot of work to refactor that.
It's in main now, see https://github.com/openhab/openhabian/actions/runs/8078761444
Can you check if the json still works with imager?

**Comment by mstormi** [comment]
About to release openHABian 1.9 which includes a third image, bookworm based.
Can you make that appear in imager, too ?
Not sure if it should be a secondary option for now because it's brand new and untested. wdyt?

**Comment by Larsen-Locke** [comment]
yes, I'm going to add it in the imager this weekend.

**Comment by Larsen-Locke** [comment]
What do you think about this:
![image](https://github.com/openhab/openhabian/assets/78228200/f9f6c843-bc6e-42fd-80d6-04dc9ff2121b)

**Comment by Larsen-Locke** [comment]
Option 2:
![image](https://github.com/openhab/openhabian/assets/78228200/43172f3a-0170-4d79-9434-fcbdb820f15e)

**Comment by mstormi** [comment]
#1 is better, lets use the available space for a little bit of marketing. 
Actually I think we should omit bullseye/bookworm names (choices are just confusing to any starter to use the imager)
and we should completely omit the bullseye 64 bit image, too. So there's one 32 and just one 64 bit image.
Put the 64 one first so most will select it.
Thing is, you should have >=2 GB if you want to run 64bit so this should be prominently mentioned.
Is there a means to offer choices based on hardware ? 

@kaikreuzer anything you prefer to see here ?

**Comment by Larsen-Locke** [comment]
So more like this?
![image](https://github.com/openhab/openhabian/assets/78228200/252f039a-fca8-4b91-b613-d87bf1ebe058)
The hardware can be specified: I chose for 64bit ["pi5-64bit", "pi4-64bit"] and for 32bit ["pi5-32bit", "pi4-32bit", "pi3-32bit"]
I didn't repeat the advertising "A vendor and technology..." because it's in the menue before anyway. What do you think?

**Comment by mstormi** [comment]
Looks good, just a minor rewording "This version requires a Pi with at least 2GB of memory."

> The hardware can be specified: I chose for 64bit ["pi5-64bit", "pi4-64bit"] and for 32bit ["pi5-32bit", "pi4-32bit", "pi3-32bit"]

Also add Pi1 and Pi2 as compatible for the 32bit image

**Comment by Larsen-Locke** [comment]
All done. PR is out. I need a release to test completely as I cannot do a build in my fork

**Comment by mstormi** [comment]
Thanks. However I don't understand what you mean by "cannot do a build".
Your PR upload resulted in an image being built, if finished you will be having your image, no?
https://github.com/openhab/openhabian/actions/runs/8132232314

**Comment by Larsen-Locke** [comment]
Yes, the image is build. But how can I get the json and check if all the links are matchin with the paths of the download folder? The only way I know is after a release.

**Comment by Larsen-Locke** [comment]
I found the json. It looks good, but I cannot do a full test until a beta-release

**Comment by kaikreuzer** [comment]
> @kaikreuzer anything you prefer to see here ?

Not really - I leave this to you guys.

**Comment by mstormi** [comment]
> I found the json. It looks good, but I cannot do a full test until a beta-release

-> 1.9beta2 release

**Comment by Larsen-Locke** [comment]
I tested both images with rpi-imager and both are written correctly, and the checksums work.

**Comment by mstormi** [comment]
thanks

**Comment by mstormi** [comment]
released 1.9
Could you please validate if that works with the imager ?

**Comment by retonap** [comment]
I have successfully downloaded and installed the OH4 64 Bit image on my Rasp4 8GB. Great job! 
After I restored the full OH backup thing were working fine (beside the Zigbee devices, because they rely on the Conbee II stick which fails during install from openhab config GUI with error 20 because of missing libqt5 but that's the same error I had when I installed the 32Bit image some month ago so it is not 64 Bit specific)

**Comment by Larsen-Locke** [comment]
The raspberry imager works for 64bit.
But no access point was started when trying to setup via wifi.
![PXL_20240321_061214743.jpg](https://github.com/openhab/openhabian/assets/78228200/3e3c7555-1501-47dd-9cac-b3ab534c6b72)

**Comment by mstormi** [comment]
Unrelated to imager or 64bit so if you want this to be a bug report, open a new issue please.
Provide debug level first-boot.log. Check if the hot spot is there after a reboot.

**Comment by mstormi** [comment]
Raspi imager is said to install old images.

@Larsen-Locke please check.

https://community.openhab.org/t/openhabian-testing/147079/109

**Comment by Larsen-Locke** [comment]
32bit is bullseye and 64bit is bookworm. I proposed more variants but @mstormi rejected.

**Comment by mstormi** [comment]
That was a misunderstanding. I've changed the json file in 1.9 release to point to bookworm-32 image.

**Comment by mstormi** [comment]
v1.9c released
